internal/report/github/summary.go

```diff
@@ -0,0 +1,127 @@
+package github
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/saucelabs/saucectl/internal/report"
+)
+
+type Reporter struct {
+	beginTime       time.Time
+	stepSummaryFile string
+	results         []report.TestResult
+}
+
+func NewJobSummaryReporter() Reporter {
+	fmt.Printf("GITHUB_STEP_SUMMARY: %s", os.Getenv("GITHUB_STEP_SUMMARY"))
+	return Reporter{
+		beginTime:       time.Now(),
+		stepSummaryFile: os.Getenv("GITHUB_STEP_SUMMARY"),
+	}
+}
+
+func (r *Reporter) isActive() bool {
+	return r.stepSummaryFile != ""
+}
+
+func (r *Reporter) Add(t report.TestResult) {
+	if !r.isActive() {
+		return
+	}
+	r.results = append(r.results, t)
+}
+
+func hasSomeDevice(results []report.TestResult) bool {
+	for _, t := range results {
+		if t.DeviceName != "" {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *Reporter) Render() {
+	if !r.isActive() {
+		return
+	}
+
+	endTime := time.Now()
+	hasDevices := hasSomeDevice(r.results)
+	errors := 0
+	inProgress := 0
+
+	content := renderHeader(hasDevices)
+	for _, result := range r.results {
+		if result.Status == "in progress" {
+			inProgress++
+		}
+		if result.Status == "failed" {
+			errors++
+		}
+		content += renderTestResult(result, hasDevices)
+	}
+	content += renderFooter(errors, inProgress, len(r.results), endTime.Sub(r.beginTime))
+
+	err := os.WriteFile(r.stepSummaryFile, []byte(content), 0x644)
+	if err != nil {
+		fmt.Printf("Unable to save summary: %v", err)
```

Comment:
That'd mess up the lines that follow:
```suggestion
		fmt.Printf("Unable to save summary: %v\n", err)
```